File: content/problems/0226-invert-binary-tree/solutions/iterative-bfs/Solution.kt
class Solution {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null) return null
        val queue = java.util.ArrayDeque<TreeNode>()
        queue.offer(root)
        while (queue.isNotEmpty()) {
            val node = queue.poll()
            node.left = node.right.also { node.right = node.left }
            node.left?.let(queue::offer)
            node.right?.let(queue::offer)
        }
        return root
    }
}

File: content/problems/0226-invert-binary-tree/solutions/iterative-bfs/solution.cpp
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (!root) return nullptr;
        queue<TreeNode*> nodes;
        nodes.push(root);
        while (!nodes.empty()) {
            TreeNode* node = nodes.front(); nodes.pop();
            swap(node->left, node->right);
            if (node->left) nodes.push(node->left);
            if (node->right) nodes.push(node->right);
        }
        return root;
    }
};
